sdk/src/transaction/mod.rs

```diff
@@ -1074,7 +1074,7 @@ impl Transaction {
     }
 }
 
-pub fn uses_durable_nonce(tx: &Transaction) -> Option<&CompiledInstruction> {
+pub fn maybe_uses_durable_nonce(tx: &Transaction) -> Option<&CompiledInstruction> {
```

Comment:
Thinking about this one more, I wonder if we can come up with a more descriptive fn name; the returned ix definitely does use durable nonce but the nonce account may not be writable.
Also, I quibble with "uses" in the first place, since this is a method that returns the instruction, not a bool.
Super wordy, but what about `try_get_durable_nonce_instruction`? Sort of ignores the "maybe writable" part

Reply:
Yeah it definitely looks like it's using a durable nonce but in the context of the Solana runtime, a tx is only considered to be using a durable nonce if it's a "valid" durable nonce which requires the nonce account to be writable. Maybe a name with "valid" in it would be more clear?

Reply:
Yeah, fair. I'm open to adding "valid" somehow

Reply:
Actually I think I was focusing too much on the validity... a nonce is also only valid if the transaction's recent blockhash is correct. This method should probably not be doing the writable check at all.

I'm inclined to go with your suggestion of `try_get_durable_nonce_instruction` but I will add it as an alias and deprecate the original function because I really shouldn't have introduced that breaking change in the first place.

Reply:
Ok actually I think I'm going to just leave everything as is but take out the is-writable check from this function